File: src/2/2.1/2.1-5/add_binary_integers.py
def add_binary_integers(a, b, n):
    result = [0] * (n + 1)
    carry = 0
    for i in range(n - 1, -1, -1):
        result[i + 1] = (a[i] + b[i] + carry) % 2
        carry = (a[i] + b[i] + carry) // 2
    result[0] = carry
    return result

File: src/2/2.1/2.1-5/add-binary-integers.ts
export function addBinaryIntegers(
  a: Array<number>,
  b: Array<number>,
  n: number
) {
  const result = Array.from<number>({ length: n + 1 }).fill(0)

  for (let i = n - 1; i >= 0; i--) {
    const sum = a[i] + b[i] + result[i + 1]
    result[i + 1] = sum % 2
    result[i] = Math.floor(sum / 2)
  }

  return result
}
